File: FC.Shared/Entities/MenuItem.cs
using FC.Interfaces.Data;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FC.Shared.Entities
{
    public class MenuItem : BaseModel
    {
        [Key]
        public Guid? MenuItemID { get; set; }
        public Guid? ParentID { get; set; }
        public Guid? OpositeID { get; set; } //example usage see PUBLISH / UNPUBLISH ITEMS.
        public string FAIcon { get; set; }
        public string Title { get; set; }
        public string OnClick { get; set; }
        /// <summary>
        /// When IsSpecific is true, this item shall not be visible when the user views a collection page. Only visible when user views detail pages like artist detail / festival detail etc.
        /// </summary>
        public bool IsSpecific { get; set; }
        [Index]
        public Guid? SectionID { get; set; }


    }
}

File: FC.WebMVC/Scripts/Shared/Models/MenuItem.ts
module FC.Shared.Models {
    export class MenuItem {
        public MenuItemID:string;
        public ParentID:string;
        public OpositeID:string;
        public SortOrder:number;
        public FAIcon:string;
        public Title:string;
        public URL:string;
        public OnClick:string;
        public IsSpecific:boolean;
        public SectionID:string;
        public Name:string;
        public Deleted:boolean;
        public IsPublished:boolean;
    }
}